### PR TITLE
🔨  Fix API call Schema types and body structure 

### PR DESCRIPTION
**Pull request by utkarsha-deriv**
This PR will help to do following: 

- Fix `Schema Header` information for api calls using oneOf in JSON structure
- This is done by creating new components `SchemaOneOfObjectContent` and `SchemaOneOfBodyHeader` to handle `oneOf` use case separately
- `SchemaOneOfBodyHeader` will display header based on the values coming from `oneOf`

- Fixes the display type of Array Items inside request/response body
- Fixes Schema display of `crypto_config` and other api calls passing `pattern` inside JSON
- Test cases for above changes and components
- Added crypto_config, crypto_estimations api calls to dropdown

### CODE REVIEW

File: src/features/Apiexplorer/Schema/RecursiveContent/SchemaBodyHeader/__tests__/SchemaBodyHeader.test.tsx [hunk 1]
```diff
@@ -74,6 +77,7 @@ describe('SchemaBodyHeader', () => {
         is_open_object
         setIsOpenObject={() => jest.fn()}
         is_stream_types={false}
+        items_type={undefined}
```
Comment (amina-deriv) [inline src/features/Apiexplorer/Schema/RecursiveContent/SchemaBodyHeader/__tests__/SchemaBodyHeader.test.tsx:80]: do we need to pass undefined to items_type?
cant we make items_type as optional in component props?

File: src/features/Apiexplorer/Schema/RecursiveContent/SchemaBodyHeader/__tests__/SchemaBodyHeader.test.tsx [hunk 2]
```diff
@@ -178,4 +187,24 @@ describe('SchemaBodyHeader', () => {
     expect(stream_types).toBeVisible();
     expect(array_type).toBeVisible();
   });
+
+  it('should render the SchemaBodyHeader for array with items type', async () => {
+    render(
+      <SchemaBodyHeader
+        key_value='test_key_value'
+        type={'array'}
```
Comment (amina-deriv) [inline src/features/Apiexplorer/Schema/RecursiveContent/SchemaBodyHeader/__tests__/SchemaBodyHeader.test.tsx:195]: type='array'

File: src/features/Apiexplorer/Schema/RecursiveContent/SchemaBodyHeader/__tests__/SchemaBodyHeader.test.tsx [hunk 3]
```diff
@@ -178,4 +187,24 @@ describe('SchemaBodyHeader', () => {
...
+        defaultValue='default_test'
+        pattern='some_test_pattern'
+        examples={['example1', 'example2']}
+        enum={['test1', 'test2']}
+        title={undefined}
+        is_open_object
+        setIsOpenObject={() => jest.fn()}
+        is_stream_types={false}
+        items_type={'string'}
```
Comment (amina-deriv) [inline src/features/Apiexplorer/Schema/RecursiveContent/SchemaBodyHeader/__tests__/SchemaBodyHeader.test.tsx:204]: items_type='string'

File: src/features/Apiexplorer/Schema/RecursiveContent/SchemaObjectContent/index.tsx [hunk 4]
```diff
@@ -74,7 +76,7 @@ export default function SchemaObjectContent({
         {!is_code_open && (
           <RecursiveProperties
             is_open={is_open_object}
-            properties={value.properties || value?.items?.properties}
+            properties={value.properties || value?.items?.properties || value.patternProperties}
```
Comment (amina-deriv) [inline src/features/Apiexplorer/Schema/RecursiveContent/SchemaObjectContent/index.tsx:79]: value?.properties

File: src/features/Apiexplorer/Schema/RecursiveContent/SchemaOneOfBodyHeader/__tests__/SchemaOneOfBodyHeader.test.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import SchemaOneOfBodyHeader from '..';
+import { screen, render } from '@testing-library/react';
+
+describe('SchemaOneOfBodyHeader', () => {
+  it('should render SchemaOneOfBodyHeader with button having object type as display name', () => {
+    render(
+      <SchemaOneOfBodyHeader
+        key_title={'cashier'}
```
Comment (amina-deriv) [inline src/features/Apiexplorer/Schema/RecursiveContent/SchemaOneOfBodyHeader/__tests__/SchemaOneOfBodyHeader.test.tsx:9]: key_title='cashier'

File: src/features/Apiexplorer/Schema/RecursiveContent/SchemaBodyHeader/index.tsx [hunk 6]
```diff
@@ -110,7 +112,7 @@ const SchemaBodyHeader = ({
                 <div className={styles.schemaObjectContent}>
                   <div>
                     <button onClick={() => setIsOpenObject(!is_open_object)}>
-                      {title ? key_value : 'object'}
+                      {title ? key_value : items_type ? items_type : 'object'}
```
Comment (amina-deriv) [inline src/features/Apiexplorer/Schema/RecursiveContent/SchemaBodyHeader/index.tsx:115]: `{title ? key_value : items_type ??  'object'}`

Comment (utkarsha-deriv) [reply src/features/Apiexplorer/Schema/RecursiveContent/SchemaBodyHeader/index.tsx:115]: Thank you 👍

**Review by matin-deriv — APPROVED** [review]

**Review by amina-deriv — APPROVED** [review]

**Pull request merged**